src/renderer/components/configure-recording-set-page/index.tsx

```diff
@@ -115,154 +47,185 @@ const ConfigureRecordingSetPage: FC<{
     getLSKey('ConfigureRecordingSetPage', 'chosenCustomListPath'),
     '',
   );
-  const setChosenBuiltInList = (name: string): void => {
-    rawSetChosenBuiltInList(name);
-    if (name) {
-      rawSetChosenCustomListPath('');
-    }
-  };
-  const setChosenCustomListPath = (filePath: string): void => {
-    rawSetChosenCustomListPath(filePath);
-    if (filePath) {
-      rawSetChosenBuiltInList('');
-    }
-  };
-
-  const addRecordingSet = async (newRecordingSet: RecordingSet): Promise<void> => {
-    if (recordingProject) {
-      const recordingSetPath = join(recordingProject.rootPath, newRecordingSet.name);
-      await ensureFolderExists(recordingSetPath);
-      setRecordingSets([...recordingSets, newRecordingSet]);
-    }
-  };
 
-  const [selectedRecordingSetIndex = -1, setSelectedRecordingSetIndex] = useLocalStorage<number>(
-    getLSKey('ConfigureRecordingSetPage', 'selectedRecordingSetIndex'),
-    -1,
-  );
+  const updatePaths = (listName: string, isBuiltIn: boolean) => {
```

Comment:
I can extract getFilePath easily.
I originally kept updatePaths in index.tsx because it uses 5 different setState functions and I didn't want to have to pass those through three layers of components, but i could just create a single function that takes in values for and sets all the values and pass that single function through?

Reply:
Yeah that's an option, or wait until you do the other refactoring I mentioned below and there may be better ways to do it.